pyvenv.el

```diff
@@ -285,9 +285,19 @@ This is usually the base name of `pyvenv-virtual-env'.")
 (defvar pyvenv-workon-history nil
   "Prompt history for `pyvenv-workon'.")
 
+(defun pyvenv--already-activated (name)
+  "Determine if the virtual environment NAME is already activated."
+  (and (boundp 'pyvenv-virtual-env-name)
+       (equal pyvenv-virtual-env-name name)))
+
 ;;;###autoload
-(defun pyvenv-workon (name)
-  "Activate a virtual environment from $WORKON_HOME."
+(defun pyvenv-workon (name &optional use-existing)
+  "Activate a virtual environment from $WORKON_HOME.
+
+If USE-EXISTING is true and the virtual environment NAME is
+already active, this function does not try to reactivate the
+environment.
```

Comment:
I think you can leave out the optional argument. The only use I can see is that it could be used to optionally disable the check with a prefix argument, but it just complicates the code.

Reply:
Are you sure nobody is relying on activation hooks to be run whenever they call `workon`? I can't think of a completely concrete example where this would be important, but we could substantially change behavior if we always skip re-activation.

Reply:
I'm fine with breaking subtle behavior if it results in cleaner code. Users who rely on this behavior can run `pyvenv-deactivate` first explicitly. (Explicit is better than implicit anyhow :-D)